internal/pkg/docker/dockerengine/dockerengine.go

```diff
@@ -339,13 +353,103 @@ func (c DockerCmdClient) Run(ctx context.Context, options *RunOptions) error {
 
 // IsContainerRunning checks if a specific Docker container is running.
 func (c DockerCmdClient) IsContainerRunning(ctx context.Context, name string) (bool, error) {
+	state, err := c.containerState(ctx, name)
+	if err != nil {
+		return false, err
+	}
+	switch state.Status {
+	case containerStatusRunning:
+		return true, nil
+	case containerStatusExited:
+		return false, &ErrContainerExited{name: name, exitcode: state.ExitCode}
+	}
+	return false, nil
+}
+
+// IsContainerCompleteOrSuccess returns true if a docker container exits with an exitcode.
+func (c DockerCmdClient) IsContainerCompleteOrSuccess(ctx context.Context, containerName string) (bool, int, error) {
```

Comment:
Is it a good idea to return `-1` when the container is still running 💭 And the method just returns the exit code.